src/app/user/profile/profile.component.ts

```diff
@@ -200,7 +200,7 @@ export class ProfileComponent extends BaseFormComponent<IUser>
   }
 
   convertTypeToPhoneType(type: string): PhoneType {
-    return $enum(PhoneType).asValueOrThrow(type)
+    return PhoneType[$enum(PhoneType).asKeyOrThrow(type)]
```

Comment:
The list of types passed into the drop-down are from `$enum(PhoneTypes).getKeys()`, so we want to verify the value we get back is a key (`asKeyOrThrow`) and then look that up in the enum.